src/aiida/transports/plugins/local.py

```diff
@@ -877,8 +877,13 @@ def rename(self, oldpath: TransportPath, newpath: TransportPath):
             raise ValueError(f'Destination {newpath} is not a valid string')
         if not os.path.exists(oldpath):
             raise OSError(f'Source {oldpath} does not exist')
-        if not os.path.exists(newpath):
-            raise OSError(f'Destination {newpath} does not exist')
+        # if not os.path.exists(newpath):
+        #     raise OSError(f'Destination {newpath} does not exist')
+
+        # Ensure the destination folder exists
+        dest_dir = os.path.dirname(newpath)
+        if dest_dir and not os.path.exists(dest_dir):
+            raise OSError(f'Destination directory {dest_dir} does not exist')
```

Comment:
Thanks for noticing this! to me it's more convenient that you do the same check as you do for `ssh.py`.
I see that you suggest to check for directory, but in principle that should not be an issue, as this method should only be used for renaming and not moving the file.
```suggestion
        if os.path.exists(newpath):
            raise OSError(f'Destination {newpath} already exists.')
```

also please update the docstring:
```
        :raises OSError: if oldpath is not found or newpath already exists.
```

Reply:
@ayushjariyal , maybe you forgot this? :)